### DISCUSSION TITLE
Format of Image for ONNX Model

### DISCUSSION THREAD

**Discussion by EdwardAndersonMcDermott** [ultralytics/yolov5#6576] [Q&A] [answered]
When we export the model to onnx, for use in a c# application, I'm not sure what processing of the image needs to happen as when we use the .pt model in python we don't have to process the image at all.

Currently we are resizing the image to match the dimensions expected by the model (stretching the image), and also normalising all of the pixel values by dividing by 255.

On the same image, the ONNX model in C# gives different results to the .pt model and ONNX model when used in python.  So we must be missing something or doing something wrong with our processing of the image.

**Comment by glenn-jocher** [comment, marked as answer]
@EdwardAndersonMcDermott simple resizing may distort the aspect ratio of the original objects. YOLOv5 uses letterboxing to maintain aspect ratio at any size, i.e. circles remain circles.

**Reply by EdwardAndersonMcDermott** [reply]
Thank you @glenn-jocher .  So if our base image size is 1280 x 720 and we need it to be 1280 x 1280, then we should pad the smaller dimension to reach 1280 rather than interpolate to create a stretched image?  I think we tried this and still got different results but I'll go double check.

**Reply by glenn-jocher** [reply]
@EdwardAndersonMcDermott yes. Either that or in your training implement distortion strategies to match what you will do in deployment.

There's no right answer, the only obvious guideline here is that your training and deployment space should be similar in order for your solution to generalize.

![Screenshot 2021-08-04 at 13 15 05](https://user-images.githubusercontent.com/26833433/153014381-a194c6de-229f-43de-b499-4a0cd8abac0d.jpg)

**Reply by EdwardAndersonMcDermott** [reply]
Indeed, my challenge is that I don't know exactly what processing the YOLOv5 algorithm does to the images.  Is there any documentation about exactly what happens to the image prior to be passed to the model for training?  We don't process the images in any way before passing them to train.py.

**Reply by glenn-jocher** [reply]
@EdwardAndersonMcDermott see common.py

https://github.com/ultralytics/yolov5/blob/f40854b61b2b347e1ff03a6fc0a765f1210a4fdc/models/common.py#L510-L535

**Reply by EdwardAndersonMcDermott** [reply]
Thank you once again @glenn-jocher.  We'll need to take some time to work through this but we now have something to go on.
